lib/basedriver/commands/find.js

```diff
@@ -47,6 +65,120 @@ commands.findElementsFromElement = async function (strategy, selector, elementId
   return await this.findElOrElsWithProcessing(strategy, selector, true, elementId);
 };
 
+helpers.findByImage = async function (b64Template, {checkingStaleness = false} = {}) {
+  let threshold = DEFAULT_MATCH_THRESHOLD;
+  if (this.settings) {
+    threshold = this.settings.getSettings().imageMatchThreshold;
+  }
+  log.info(`Finding image element with match threshold ${threshold}`);
+  const {width: screenWidth, height: screenHeight} = await this.getWindowSize();
+
+  // someone might have sent in a template that's larger than the screen
+  // dimensions. If so let's check and cut it down to size since the algorithm
+  // will not work unless we do
+  b64Template = await this.ensureTemplateSize(b64Template, screenWidth, screenHeight);
```

Comment:
I would rather skip this check and let the client to be responsible for template size. It might be too expensive to call `getWindowSize` each time the lookup is executed.